### PR TITLE
map.featuresIn example

### LINKED ISSUE #1494: featuresIn example
From https://github.com/mapbox/mapbox-gl-js/issues/1264#issuecomment-141074829:

> Any example on how to replace the default <kbd>shift</kbd>+<kbd>drag</kbd> **zoom** with **selection** of all markers inside that area.

### PR DESCRIPTION
**Pull request by tristen**
featuresin-example

cc @jfirebaugh for review

- Closes #1494

### CODE REVIEW

File: docs/_posts/examples/3400-01-19-using-featuresin.html [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    function mousePos(e) {
+        var rect = canvas.getBoundingClientRect();
+        return new mapboxgl.Point(
+            e.clientX - rect.left - canvas.clientLeft,
+            e.clientY - rect.top - canvas.clientTop
+        );
+    }
+
+    function mouseDown(e) {
+        // Continue the rest of the function if the shitkey is pressed.
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-19-using-featuresin.html]: Nice typo! :hankey:

File: docs/_posts/examples/3400-01-19-using-featuresin.html [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        // Disable default drag zooming when the shift key is held down.
+        map.dragPan.disable();
+
+        // Call functions for the following events
+        document.addEventListener('mousemove', onMouseMove);
+        document.addEventListener('mouseup', onMouseUp);
+        document.addEventListener('keydown', onKeyDown);
+
+        // Capture the fist xy coordinates
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-19-using-featuresin.html]: first

File: docs/_posts/examples/3400-01-19-using-featuresin.html [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            map.getCanvas().style.cursor = (!err && features.length) ? 'pointer' : '';
+
+            if (err || !features.length) {
+                popup.remove();
+                return;
+            }
+
+            var feature = features[0];
+
+            popup.setLngLat([e.lngLat.lng, e.lngLat.lat])
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-19-using-featuresin.html]: `popup.setLngLat(e.lngLat)`

**Comment by jfirebaugh** [comment]
Instead of adding two sources and calling `setData` on one of them with data returned by `featuresIn`, the example should use a single source and update filters. The reason is that `featuresIn` will return features chopped up into tiles, and the example as written will behave unexpectedly around tile borders.

File: docs/_posts/examples/3400-01-19-using-featuresin.html (hunk 1 above)
Comment (tristen) [inline docs/_posts/examples/3400-01-19-using-featuresin.html]: LOL

**Comment by tristen** [comment]
> Instead of adding two sources and calling setData on one of them with data returned by featuresIn, the example should use a single source and update filters. The reason is that featuresIn will return features chopped up into tiles, and the example as written will behave unexpectedly around tile borders.

Gotcha. How's this approach? https://github.com/mapbox/mapbox-gl-js/commit/09417173bea5f9ae1d76430eef2bc5741c2ad322

File: docs/_posts/examples/3400-01-19-using-featuresin.html [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+        if (box) {
+            box.parentNode.removeChild(box);
+            box = null;
+        }
+
+        // If bbox exists. use this value as the argument for `featuresIn`
+        if (bbox) {
+            map.featuresIn(bbox, {
+                includeGeometry: true,
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-19-using-featuresin.html]: Don't need geometry included now.

File: docs/_posts/examples/3400-01-19-using-featuresin.html [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+            if (err || !features.length) {
+                popup.remove();
+                return;
+            }
+
+            var feature = features[0];
+
+            popup.setLngLat(e.lngLat)
+                .setHTML(feature.properties.COUNTY)
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-19-using-featuresin.html]: `setText(feature.properties.COUNTY)`

**Comment by jfirebaugh** [comment]
:+1:

**Comment by jakepruitt** [comment]
:heart_eyes_cat:

**Pull request merged**